File: androidApp/src/main/java/com/asif/kmmauthorizedimageprofiles/android/home/HomeViewModel.kt
package com.asif.kmmauthorizedimageprofiles.android.home

import androidx.lifecycle.ViewModel
import com.asif.kmmauthorizedimageprofiles.android.SecurePreferences

class HomeViewModel(private val securePreferences: SecurePreferences) : ViewModel() {

    fun logout() {
        securePreferences.clearToken()
    }
}

File: iosApp/iosApp/HomeViewModel.swift
//
//  HomeViewModel.swift
//  iosApp
//
//  Created by AsifMacMini on 23/08/24.
//  Copyright © 2024 orgName. All rights reserved.
//

import Foundation

/// A SwiftUI ObservableObject ViewModel for managing the home screen logic, including logout.
class HomeViewModel: ObservableObject {
    private let securePreferences: SecurePreferences

    /// Initializes the ViewModel with required dependencies.
    init(securePreferences: SecurePreferences) {
        self.securePreferences = securePreferences
    }

    /// Logs the user out by clearing the stored token.
    func logout() {
        securePreferences.clearToken()
    }
}
